conans/client/rest/conan_requester.py

```diff
@@ -123,9 +114,21 @@ def __init__(self, config, cache_folder=None):
                                    platform.machine()])
         self._user_agent = "Conan/%s (%s)" % (__version__, platform_info)
 
-    @staticmethod
-    def _get_retries(config):
-        retry = config.get("core.net.http:max_retries", default=2, check_type=int)
+    @property
+    def _http_requester(self):
+        if ConanRequester._http_requester_imp is None:
+            # FIXME: Trick for testing when requests is mocked
+            if hasattr(requests, "Session"):
+                ConanRequester._http_requester_imp = requests.Session()
+                adapter = HTTPAdapter(max_retries=self._get_retries())
```

Comment:
But this will only get the ``max_retries`` of the first invocation, if it changes later the conf, it will be ignored.
It would be good to check if caching just the session but redefining the HTTPAdapter also works.

Reply:
If we reset the mounts, all previous versions will also use the new retry. I think that's a better behaviour than staying with the first one, good idea